airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -119,18 +124,24 @@ const getOptimalSyncMode = (
 const calculateInitialCatalog = (
   schema: SyncSchema,
   supportedDestinationSyncModes: DestinationSyncMode[],
-  isNotCreateMode?: boolean
-): SyncSchema => ({
-  streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
-    const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
-    const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
-
-    if (isNotCreateMode) {
-      return nodeStream;
-    }
-
-    return getOptimalSyncMode(verifyConfigCursorField(nodeStream), supportedDestinationSyncModes);
-  }),
-});
+  isNotCreateMode?: boolean,
+  newStreamDescriptors?: StreamDescriptor[]
+): SyncSchema => {
+  return {
+    streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
+      const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
+      const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
+
+      // if the stream is new since a refresh, we want to verify cursor and get optimal sync modes
+      const matches = newStreamDescriptors?.filter(
```

Comment:
If you're just interested whether any new stream descriptions match, maybe `array.some()` would be better than `array.filter()`? That would make the comparison on line 139 a little less verbose: `if (isNotCreateMode && !matches)`